modules/47_regipol/regiCarbonPrice/bounds.gms

```diff
@@ -18,41 +18,122 @@
 *'
 *' ###### Bounds for Historic and Near-term Alignment
 
+*' ####### Power Sector
+
+$ifThen.tech_bounds_2025 "%cm_tech_bounds_2025%" == "on"
 *' This limits wind and solar PV capacity additions for 2025 in light of recent slow developments as of 2023.
 *' Upper bound is double the historic maximum capacity addition in 2011-2020.
+*' In addition: Limit solar PV capacity to 120 GW in 2025 (2023-2027 average) given that we are at only 76 GW PV in 2023
 loop(regi$(sameAs(regi,"DEU")),
   vm_deltaCap.up("2025",regi,"wind","1")=2*smax(tall$(tall.val ge 2011 and tall.val le 2020), pm_delta_histCap(tall,regi,"wind"));
   vm_deltaCap.up("2025",regi,"spv","1")=2*smax(tall$(tall.val ge 2011 and tall.val le 2020), pm_delta_histCap(tall,regi,"spv"));
+  vm_cap.up("2025",regi,"spv","1")=0.12;
 );
+$endIf.tech_bounds_2025
+
 
 *' These bounds account for historic gas power development.
-vm_capTotal.up("2015",regi,"pegas","seel")$(sameas(regi,"DEU"))=30/1000;
-vm_capTotal.up("2020",regi,"pegas","seel")$(sameas(regi,"DEU"))=34/1000;
+v47_prodSEtotal.up("2020",regi,"pegas","seel")$(sameAs(regi,"DEU"))= 0.36*sm_EJ_2_TWa;
+v47_prodSEtotal.up("2025",regi,"pegas","seel")$(sameAs(regi,"DEU"))= 0.4*sm_EJ_2_TWa;
 
 *' These bounds account for historic coal power development.
 vm_cap.up("2020",regi,"pc","1")$((cm_startyear le 2020) and (sameas(regi,"DEU"))) = 38.028/1000;
+*' This limits early retirement of coal power in Germany in 2020s to avoid extremly fast phase-out.
+vm_capEarlyReti.up('2025',regi,'pc')$(sameAs(regi,"DEU")) = 0.65;
 
-*' These bounds account for historic and near-term gas power development and prevent too sudden coal to gas switch in Germany by 2025.
-loop(regi$(sameAs(regi,"DEU")),
-vm_deltaCap.up("2015",regi,"ngcc","1") = 0.002;
-vm_deltaCap.up("2020",regi,"ngcc","1") = 0.0015;
-vm_deltaCap.up("2025",regi,"ngcc","1") = 0.0015;
-*' Along the same lines, this limits early retirement of coal power in Germany in 2020s to avoid extremly fast phase-out.
-vm_capEarlyReti.up('2025',regi,'pc') = 0.65; 
-);
+*' DEU coal-power capacity phase-out, upper bounds following the Kohleausstiegsgesetz from 2020.
+*' https://www.bmuv.de/faqs/kohleausstiegsgesetz
+    vm_capTotal.up("2025",regi,"pecoal","seel")$(sameas(regi,"DEU"))=25/1000;
+    vm_capTotal.up("2030",regi,"pecoal","seel")$(sameas(regi,"DEU"))=17/1000;
+    vm_capTotal.up("2035",regi,"pecoal","seel")$(sameas(regi,"DEU"))=6/1000;
+    vm_capTotal.up("2040",regi,"pecoal","seel")$(sameas(regi,"DEU"))=1.1/1000;
 
 
-*' This limits coal-power capacity to at least 5 GW in 2030 to account for emissions 
```

Comment:
Yes. Since the feedstocks implementation we have [waste emissions](https://github.com/remindmodel/remind/blob/901051edbfe13645207d7470632e3e000730bbd4/core/equations.gms#L613) in the model so we do not need to artificially add some capacities here.